cli/src/main/kotlin/utils/PackageConfigurationOption.kt

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+package org.ossreviewtoolkit.utils
+
+import org.ossreviewtoolkit.model.utils.PackageConfigurationProvider
+import org.ossreviewtoolkit.model.utils.SimplePackageConfigurationProvider
+
+internal sealed class PackageConfigurationOption {
```

Comment:
This trick with the sealed class is only required if the different mutually exclusive options are of different types (like `File` and `String`), but this is not the case here. For the common simple case where all options are of the same type, you can do it like in `ScannerCommand` and get rid of this file:

https://github.com/oss-review-toolkit/ort/blob/b8bbfa0bfb50fdb2de62e70bb03300e352ce9d3c/cli/src/main/kotlin/commands/ScannerCommand.kt#L54-L66

Reply:
I'm aware of this and actually this was how I had written it in the first place.
I liked the sealed class approach more because it lead to a bit more shared code and I found the case distinction (see my when clause) a tad nicer / cleaner. 

I'd prefer to keep it as is. Are you ok with that?

Reply:
Ok.